app/services/SubmitExplainDelayService.scala

```diff
@@ -52,39 +51,38 @@ class SubmitExplainDelayService @Inject()(submitExplainDelayConnector: SubmitExp
 
   def submit(ern: String, arc: String)(implicit hc: HeaderCarrier, dataRequest: DataRequest[_]): Future[SubmitExplainDelayResponse] = {
 
-    val submission = SubmitExplainDelayModel(dataRequest.movementDetails)(dataRequest.userAnswers)
+    val submissionRequest = SubmitExplainDelayModel(dataRequest.movementDetails)(dataRequest.userAnswers)
 
-    val auditCorrelationId = UUID.randomUUID().toString
-
-    auditingService.audit(
-      SubmitExplainDelayAuditModel(
-        credentialId = dataRequest.request.request.credId,
-        internalId = dataRequest.internalId,
-        correlationId = auditCorrelationId,
-        submission = submission,
-        ern = ern
-      )
-    )
-
-    submitExplainDelayConnector.submit(ern, submission).map {
-      case Right(success) =>
+    submitExplainDelayConnector.submit(ern, submissionRequest).map {
```

Comment:
There's one issue with this, there's no recover block further down the chain that I can see. So there's a chance that if an exceptions is thrown _(for example, if the connection times out and is severed)_ then no Audit Event would be logged.

We either need to:
- Introduce a recover block within the connect that will handle any thrown exceptions and return a `Left`
- Add the recover block to this service, and also handle an audit event within that block

I think my preference would be to catch it at source and do it within the connector, to ensure that it will always return an `Either[L,R]` back to the service.

Reply:
Pushed another commit to introduce the recover block within the connector.